sparse/client.go

```diff
@@ -79,41 +72,17 @@ func syncFile(localPath string, addr TCPEndPoint, remotePath string, timeout int
 	encoder := gob.NewEncoder(conn)
 	decoder := gob.NewDecoder(conn)
 
-	// Use unix time as hash salt
-	salt := make([]byte, binary.MaxVarintLen64)
-	binary.PutVarint(salt, time.Now().UnixNano())
-	status := sendSyncRequest(encoder, decoder, remotePath, size, salt)
-	if !status {
-		err = fmt.Errorf("Sync request to %v failed", remotePath)
-		log.Error(err)
+	ack, err := sendSyncRequest(encoder, decoder, remotePath, fileSize)
+	if err != nil || ack == false {
+		log.Errorf("Sync request to %s failed, error: %s, ack: %v", remotePath, err, ack)
 		return nil, err
 	}
 
-	abortStream := make(chan error)
-	layoutStream := make(chan FileInterval, 128)
-	errStream := make(chan error)
-
-	// Initiate interval loading...
-	err = loadFileLayout(abortStream, file, layoutStream, errStream)
+	err = syncFileContent(encoder, decoder, file, fileSize)
 	if err != nil {
-		log.Error("Failed to retrieve local file layout:", err)
-		return nil, err
+		log.Error("syncFileContent failed: ", err)
```

Comment:
In fact it's mostly caller's job to log this error, and you can `return nil, fmt.Errorf("syncFileContent failed: ", err)` if you want to add more details.